main/src/hadd.cxx

```diff
@@ -90,281 +99,382 @@
-   int outputPlace = 0;
-   int ffirst = 2;
-   Int_t newcomp = -1;
-   for( int a = 1; a < argc; ++a ) {
-      if ( strcmp(argv[a],"-T") == 0 ) {
-         noTrees = kTRUE;
-         ++ffirst;
-      } else if ( strcmp(argv[a],"-a") == 0 ) {
-         append = kTRUE;
-         ++ffirst;
-      } else if ( strcmp(argv[a],"-f") == 0 ) {
-         force = kTRUE;
-         ++ffirst;
-      } else if ( strcmp(argv[a],"-k") == 0 ) {
-         skip_errors = kTRUE;
-         ++ffirst;
-      } else if ( strcmp(argv[a],"-O") == 0 ) {
-         reoptimize = kTRUE;
-         ++ffirst;
-      } else if (strcmp(argv[a], "-dbg") == 0) {
-         debug = kTRUE;
-         verbosity = kTRUE;
-         ++ffirst;
-      } else if (strcmp(argv[a], "-d") == 0) {
-         if (a + 1 != argc && argv[a + 1][0] != '-') {
-            if (gSystem->AccessPathName(argv[a + 1])) {
-               std::cerr << "Error: could not access the directory specified: " << argv[a + 1]
-                         << ". We will use the system's temporal directory.\n";
-            } else {
-               workingDir = argv[a + 1];
-            }
-            ++a;
-            ++ffirst;
-         } else {
-            std::cout << "-d: no directory specified.  We will use the system's temporal directory.\n";
-         }
-         ++ffirst;
-      } else if (strcmp(argv[a], "-j") == 0) {
-         // If the number of processes is not specified, use the default.
-         if (a + 1 != argc && argv[a + 1][0] != '-') {
-            // number of processes specified
-            Long_t request = 1;
-            for (char *c = argv[a + 1]; *c != '\0'; ++c) {
-               if (!isdigit(*c)) {
-                  // Wrong number of Processes. Use the default:
-                  std::cerr << "Error: could not parse the number of processes to run in parallel passed after -j: "
-                            << argv[a + 1] << ". We will use the system maximum.\n";
-                  request = 0;
-                  break;
-               }
-            }
-            if (request == 1) {
-               request = strtol(argv[a + 1], 0, 10);
-               if (request < kMaxLong && request >= 0) {
-                  nProcesses = (Int_t)request;
-                  ++a;
-                  ++ffirst;
-                  std::cout << "Parallelizing  with " << nProcesses << " processes.\n";
+   int size;
+   auto parseResult = ROOT::FromHumanReadableSize(arg, size);
+   if (parseResult == ROOT::EFromHumanReadableSize::kParseFail) {
+      std::cerr << "Error: could not parse the cache size passed after -cachesize: " << arg
+                << ". We will use the default value.\n";
+      return {};
+   } else if (parseResult == ROOT::EFromHumanReadableSize::kOverflow) {
+      double m;
+      const char *munit = nullptr;
+      ROOT::ToHumanReadableSize(INT_MAX, false, &m, &munit);
+      std::cerr << "Error: the cache size passed after -cachesize is too large: " << arg << " is greater than " << m
```

Comment:
Sometimes I see the wording `Error` and some other times there is `[err]`, probably we should choose one and stick to it. In other places in ROOT, although sadly not everywhere, the name of the function is used to begin the error message, like:

```
hadd: the cache size passed...
````

Reply:
I wanted to explicitly distinguish [err] and [warn] since the first cause hadd to abort and the latter doesn't.
As for the existing error messages, I was a bit afraid they would be used in some roottest so I didn't change them, but I guess it's worth doing it.
I would propose something like `hadd: [Error] the cache size...` (though in this case it's actually a warning)

Reply:
Yes I agree with having some different wording for error and warnings. Your proposal looks fine to me :+1:

Reply:
Note that we already have `Error` and `Warning` macros available form `TError.h`

Reply:
@vepadulano the problem is that it would require rewriting all the stream-style messages to printf-style, which is kinda annoying...

Reply:
I agree with you, then let me make a proposal: it would be enough to keep the same formatting of the output, as in:
```
root [3] Error("hadd", "error message")
Error in <hadd>: error message
root [4] Warning("hadd", "warning message")
Warning in <hadd>: warning message
```
At least we would keep the consistency of the user experience.